test/distributed/elastic/agent/server/test/local_elastic_agent_test.py

```diff
@@ -963,6 +1039,56 @@ def test_double_agent_fault_tolerance_etcd_v2(self):
             backend="etcd-v2", test_to_run=self.double_agent_fault_tolerance
         )
 
+    def double_agent_clean_exit(self):
+        """
+        start ``nnodes`` agents, kill and restart odd ones, validate fault-tolerance works
+        """
+        nnodes = 2
+        wait = 20
+        node_conf = Conf(
+            entrypoint=_bipolar_sleep_function, args=(wait,), local_world_size=2
+        )
+        agent_results = mp.Queue()
+        agent_args = {
+            "conf": node_conf,
+            "agent_results": agent_results,
+            "min_nodes": nnodes,
+            "max_nodes": nnodes,
+            "max_restarts": 0,
+            "exit_barrier_timeout": 300,
+        }
+
+        procs = []
+        for _ in range(nnodes):
+            p = mp.Process(
+                target=self.run_agent,
+                kwargs=agent_args,
+            )
+            procs.append(p)
+            p.start()
+
+        # wait for all processes to finish should not take exit barrier time
+        time_between_processes = 0
+        for i in range(nnodes):
+            p = procs[i]
+            p.join()
+            self.assertNotEqual(0, p.exitcode)
+            time_between_processes = time.monotonic() - time_between_processes
+        # Validate that the processes finish close to each other.
+        # Using a slightly higher timeout than 2 * monitor_interval to make it less flaky
+        self.assertGreater(
+            1, time_between_processes, "Agents are not cleaned up until 2 * monitor_interval"
```

Comment:
shouldn't this be a function of `monitor_interval`? something like:

```
self.assertGreater(2*monitor_interval, exit_interval_between_agents)
```

Reply:
So, i ran some tests and i did get numbers greater than 2 * monitor_interval with my code, monitor_interval is 0.01 , whereas i was getting values of 0.2-0.3 seconds exit_interval_between_agents . 
Also, there is no relevant code i found around 2* monitor_interval guarantee. 
https://github.com/pytorch/pytorch/blob/main/torch/distributed/elastic/agent/server/api.py#L701

Reply:
cc @kiukchung to take a final look , i have used 2 * 0.1 as a check now, but the monitor_interval in the test is 0.01

Reply:
thanks, I'd say you plumb the `monitor_interval` in  `run_agent()` all the way through to the `get_worker_spec()` function so that you can set the monitor_interval to 0.1 explicitly without relying on the 0.01 default.